php/php-rr-sdk.md

```diff
@@ -186,41 +169,46 @@ use Temporal\Workflow;
 
 class UploadWorkflow
 {
-    public function run(Workflow\ContextInterface $ctx, string $input): string
+    public function run(string $input): string
     {
-        $promise = yield new Workflow\ExecuteActivy(
+        $promise = new Workflow\ExecuteActivy(
             'activityName',
             $input, 
-            100 // startToFinishTimeout, etc. 
+            [
+                ExecuteActivy::START_TO_FINISH_TIMEOUT   => 100,        
+                ExecuteActivy::SCHEDULE_TO_START_TIMEOUT => 10
+            ]       
         );        
     
+        // blocking operation to get the activity result
         $result = yield $promise->get(Obj::class);
 
         return $result->getValue();
     }
 }
 ```
 
-Read more about proposed syntax approach and syntax sugar down below. The given example can be simplified into:
+> Activity Stub and builder is currently out of the scope, in first version of library parameters must be passed
+> explicitly. All following examples _assume_ existence some form of domain specific activity builder which wraps command
+> construction. 
+
+### Context
+In order to ensure easy and reliable access to workflow context (including library implementations) we introduce static
+methods via `Workflow` class. Since only one workflow can be executed at one moment of time we can safely change the
+context before passing control to the workflow:
 
 ```php
-use Temporal\Workflow;
+// explicit
 
-class UploadWorkflow extends Workflow\Workflow
-{
-    public function run(string $input): string
-    {
-        $result = yield $this->activities
-            ->withStartToFinish(100)
-            ->activityName($input)
-            ->get(Obj::class);
+// implicit 
+Workflow::ctx()->now();
+Workflow::ctx()->isReplying();
 
-        return $result->getValue();
-    }
-}
+// sugar
+Workflow::executeActivity();
```

Comment:
I like more

Workflow::now();
Workflow::isReplaying();

Reply:
agreed since we moved to static context per thread